control-plane/agents/src/bin/core/tests/volume/switchover.rs

```diff
@@ -2,22 +2,180 @@
+// Starts node which was stopped and expects old nexus cleanup.
+#[tokio::test]
+async fn old_nexus_delete_after_vol_destroy() {
+    let reconcile_period = Duration::from_secs(1);
+    let cluster = ClusterBuilder::builder()
+        .with_rest(true)
+        .with_agents(vec!["core"])
+        .with_io_engines(3)
+        .with_tmpfs_pool(POOL_SIZE_BYTES)
+        .with_cache_period("1s")
+        .with_reconcile_period(reconcile_period, reconcile_period)
+        .build()
+        .await
+        .unwrap();
+
+    let vol_client = cluster.grpc_client().volume();
+    let nexus_client = cluster.grpc_client().nexus();
+
+    let vol = vol_client
+        .create(
+            &CreateVolume {
+                uuid: "ec4e66fd-3b33-4439-b504-d49aba53da26".try_into().unwrap(),
+                size: 5242880,
+                replicas: 2,
+                ..Default::default()
+            },
+            None,
+        )
+        .await
+        .expect("unable to create volume");
+
+    let vol = vol_client
+        .publish(
+            &PublishVolume {
+                uuid: vol.uuid().clone(),
+                share: None,
+                target_node: Some(cluster.node(0)),
+                publish_context: HashMap::new(),
+                frontend_nodes: vec![cluster.node(1).to_string()],
+            },
+            None,
+        )
+        .await
+        .expect("failed to publish volume");
+
+    let target_node = vol.state().target.unwrap().node;
+    assert_eq!(target_node, cluster.node(0));
+
+    let vol = vol_client
+        .republish(
+            &RepublishVolume {
+                uuid: vol.uuid().clone(),
+                target_node: Some(cluster.node(1)),
+                share: VolumeShareProtocol::Nvmf,
+                reuse_existing: false,
+                frontend_node: cluster.node(1),
+                reuse_existing_fallback: false,
+            },
+            None,
+        )
+        .await
+        .expect("unable to republish volume");
+
+    let republished_node = vol.state().target.unwrap().node;
+    assert_ne!(
+        republished_node, target_node,
+        "expected nexus node to change post republish"
+    );
+
+    let nexuses = nexus_client
+        .get(Filter::None, None)
+        .await
+        .expect("Could not list nexuses");
+    assert_eq!(nexuses.0.len(), 2, "expected two nexuses after republish");
+
+    cluster
+        .composer()
+        .stop(cluster.node(0).as_str())
+        .await
+        .expect("failed to stop container");
+
+    let nexuses_after_pause = nexus_client
+        .get(Filter::None, None)
+        .await
+        .expect("could not list nexuses");
+    sleep(Duration::from_secs(2)).await;
+
+    assert_eq!(
+        nexuses_after_pause.0.len(),
+        1,
+        "expected 1 nexus after stop"
+    );
+    let spec = cluster
+        .rest_v00()
+        .specs_api()
+        .get_specs()
+        .await
+        .expect("expected to retrieve specs");
```

Comment:
Get load specs api into var to avoid these long statements:
```suggestion
    let client = cluster.rest_v00();
    let specs_api = client.specs_api();

    let spec = specs_api().get_specs().await.unwrap();
```
Then reuse specs_api in the code..

Reply:
Done